### PR TITLE
Hook Unit Tests

### PR DESCRIPTION
**Pull request by nca6109**
Added unit tests for hook controller functions

**[Hook Update]: Changed type of `next_panel_set_id` property from `number` to `number | null` since the property can be initialized with a `null` value**

### CODE REVIEW

**Review by wct9667 — APPROVED** [review]
Looks good.

**Review by BlckHawker — CHANGES_REQUESTED** [review]
This is to block merging since I see some things that should be changed. I will write a detailed review now

File: src/tests/hook.test.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        expect(response).toEqual(hookData);
+    });
+
+    test('If the hook does not exists, it should return undefined', async () => {
+        (hookService.getHook as jest.Mock).mockReturnValue(() => Promise.resolve(undefined));
+
+        const response = await _getHookController(sequelizeMock())(1);
+
+        expect(response).toBeUndefined();
+    });
```
Comment (BlckHawker) [inline src/tests/hook.test.ts]: These are duplicate tests. It is not the controller's job to care about what is returned from the service method, but to return it.

These descriptions are inaccurate as well.

File: src/tests/hook.test.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        }
+    );
+
+    test('If the panel has no associated hooks, return empty array', async () => {
+        const panelHookData = [] as number[];
+        const panelData = {
+            image:        '../phonyImage.png',
+            index:        0,
+            panel_set_id: 1
+        };
```
Comment (BlckHawker) [inline src/tests/hook.test.ts]: Duplicate tests

File: src/tests/hook.test.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+        (panelService.getPanel as jest.Mock).mockReturnValue(() => Promise.resolve(panelData));
+        (hookService.createHook as jest.Mock).mockReturnValue(() => Promise.resolve(hookData));
+
+        const response = await _createHookController(sequelizeMock())([1, 1], 1, null);
+
+        expect(response).toBe(hookData);
+    });
```
Comment (BlckHawker) [inline src/tests/hook.test.ts]: duplicate tests

**Review by BlckHawker — CHANGES_REQUESTED** [review]
Remove some duplicate tests and revise the test descriptions, and you should be good to go. You can review the panel set tests for reference

**Review by BlckHawker — APPROVED** [review]
sonic-boom-knuckles

**Pull request merged**